signxml/verifier.py

```diff
@@ -196,8 +196,13 @@ def _apply_transforms(self, payload, *, transforms_node: etree._Element, signatu
             transforms = self._findall(transforms_node, "Transform")
 
         for transform in transforms:
-            if transform.get("Algorithm") == SignatureConstructionMethod.enveloped.value:
+            algorithm = transform.get("Algorithm")
+            if algorithm == SignatureConstructionMethod.enveloped.value:
                 _remove_sig(signature, idempotent=True)
+            if algorithm == TransformAlgorithm.dsig_filter2.value:
+                transform_xpath = self._find(transform, f"dsig_filter2:XPath")
+                if transform_xpath is not None and transform_xpath.get("Filter") == "subtract" and signature == payload.xpath(transform_xpath.text, namespaces=namespaces)[0]:
+                    _remove_sig(signature, idempotent=True)
```

Comment:
I guess we could do something like this, if we would intend to upstream it. Haven't read the spec, but it seems like it would give us a pretty complete support for the "subtract" filter at least.
```suggestion
                if transform_xpath is not None and transform_xpath.get("Filter") == "subtract":
                    elements = payload.xpath(transform_xpath.text, namespaces=namespaces)
                    for element in elements:
                        # "_remove_sig" is a bit misleading name for this func. 
                        # It is generic and just removes a node from the tree.
                        _remove_sig(element, idempotent=True)
```
But if we intend to keep it to ourselves, what we have in this PR should be more than enough for our use case, so perhaps no need to over-engineer it.

Reply:
I like this better, it conforms better to the spec. I'm already implementing a subset of it so I much rather do it properly :D